src/Microsoft.DotNet.Build.CloudTestTasks/UploadToAzure.cs

```diff
@@ -1,96 +1,229 @@
-        [Required]
-        public string AccountName { get; set; }
+        private CancellationTokenSource tokenSource;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="UploadToAzure" /> class.
+        /// </summary>
+        public UploadToAzure()
+        {
+            this.Overwrite = false;
+        }
 
         /// <summary>
-        /// The Azure account key used when creating the connection string.
+        ///     The Azure account key used when creating the connection string.
         /// </summary>
         [Required]
         public string AccountKey { get; set; }
 
         /// <summary>
-        /// The name of the container to access.  The specified name must be in the correct format, see the
-        /// following page for more info.  https://msdn.microsoft.com/en-us/library/azure/dd135715.aspx
+        ///     The Azure account name used when creating the connection string.
+        /// </summary>
+        [Required]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        ///     The name of the container to access.  The specified name must be in the correct format, see the
+        ///     following page for more info.  https://msdn.microsoft.com/en-us/library/azure/dd135715.aspx
         /// </summary>
         [Required]
         public string ContainerName { get; set; }
 
         /// <summary>
-        /// An item group of files to upload.  Each item must have metadata RelativeBlobPath
-        /// that specifies the path relative to ContainerName where the item will be uploaded.
+        ///     An item group of files to upload.  Each item must have metadata RelativeBlobPath
+        ///     that specifies the path relative to ContainerName where the item will be uploaded.
         /// </summary>
         [Required]
         public ITaskItem[] Items { get; set; }
 
         /// <summary>
-        /// Indicates if the destination blob should be overwritten if it already exists.  The default if false.
+        ///     Indicates if the destination blob should be overwritten if it already exists.  The default if false.
         /// </summary>
         public bool Overwrite { get; set; }
 
-        public UploadToAzure()
+        /// <summary>
+        /// The cancel.
+        /// </summary>
+        public void Cancel()
         {
-            Overwrite = false;
+            this.tokenSource.Cancel();
         }
 
+        /// <summary>
+        ///     The execute.
+        /// </summary>
+        /// <returns>
+        ///     The <see cref="bool" />.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// </exception>
         public override bool Execute()
         {
-            Log.LogMessage(MessageImportance.High, "Begin uploading blobs to Azure account {0} in container {1}.", AccountName, ContainerName);
+            this.tokenSource = new CancellationTokenSource();
+            var ct = this.tokenSource.Token;
```

Comment:
I'd also make this variable set in the initialize, so that you don't have a race condition of someone cancelling it before this line runs.